invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Schemaorg based Schema for Invenio RDM Records."""
+from typing import Optional
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from flask_resources.serializers import BaseSerializerSchema
+from pydash import py_
+import pycountry
+from idutils import normalize_orcid, normalize_ror
+from commonmeta import (
+    unwrap,
+    wrap,
+    parse_attributes,
+    dict_to_spdx,
+    doi_as_url,
+)
+from marshmallow import Schema, fields, missing
+from marshmallow_utils.fields import SanitizedUnicode
+
+from ..schemas import CommonFieldsMixin
+from ..utils import get_vocabulary_props
+
+
+class PersonOrOrgSchema(Schema):
+    """Creator/contributor schema for Schemaorg."""
+
+    include = {
+        "@id": fields.Method("get_name_identifier"),
+        "@type": fields.Method("get_name_type"),
+    }
+
+    name = fields.Str(attribute="person_or_org.name")
+    givenName = fields.Str(attribute="person_or_org.given_name")
+    familyName = fields.Str(attribute="person_or_org.family_name")
+    affiliation = fields.Method("get_affiliation")
+
+    def get_name_type(self, obj):
+        """Get name type. Schemaorg expects either @Person or @Organisation.
+        Defaults to @Person.
+        """
+        name_type = py_.get(obj, "person_or_org.type")
+        return "@Organisation" if name_type == "organizational" else "@Person"
+
+    def get_name_identifier(self, obj):
+        """Get name identifier. Schemaorg expects a URL for the identifier, and
+        does not support multiple identifiers. Use the first identifier found.
+        """
+
+        def format_name_identifier(name_identifier) -> Optional[str]:
+            """format_name_identifier"""
+            if name_identifier.get("identifier", None) is None:
+                return None
+            if name_identifier.get("scheme", None) == "orcid":
+                return normalize_orcid(name_identifier.get("identifier", None))
+            if name_identifier.get("scheme", None) == "ror":
+                return normalize_ror(name_identifier.get("identifier", None))
+            if name_identifier.get("scheme", None) == "isni":
+                return "http://isni.org/isni/" + name_identifier.get("identifier", None)
+            # TODO: normalize other personororg identifiers, e.g. GND, etc.
```

Comment:
Maybe we can create a ticket for the missing identifiers. What do you think?

Reply:
There needs to be some discussion of what InvenioRDM should support, as this would otherwise become a long list.